cli/azd/pkg/project/project_config.go

```diff
@@ -263,6 +264,40 @@ func ParseProjectConfig(yamlContent string, env *environment.Environment) (*Proj
 		if svc.Language == "" || svc.Language == "csharp" || svc.Language == "fsharp" {
 			svc.Language = "dotnet"
 		}
+
+		if svc.Language == "dotnet" {
+			stat, err := os.Stat(svc.Path())
+			if err != nil {
+				return nil, fmt.Errorf("stating project %s: %w", svc.Path(), err)
+			} else if stat.IsDir() {
+				entries, err := os.ReadDir(svc.Path())
+				if err != nil {
+					return nil, fmt.Errorf("listing files for service %s: %w", svc.Name, err)
+				}
+				var projectFile string
+				for _, entry := range entries {
+					switch strings.ToLower(filepath.Ext(entry.Name())) {
+					case ".csproj", ".fsproj", ".vbproj":
+						if projectFile != "" {
+							// we found multiple project files, we need to ask the user to specify which one
+							// corresponds to the service.
+							return nil, fmt.Errorf("multiple .NET project files detected in %s for service %s, please include the name of the .NET project file in 'project' setting in %s for this service", svc.Path(), svc.Name, azdcontext.ProjectFileName)
+						} else {
+							projectFile = entry.Name()
+						}
+					}
+				}
+				if projectFile == "" {
+					return nil, fmt.Errorf("could not determine the .NET project file for service %s, please include the name of the .NET project file in project setting in %s for this service", svc.Name, azdcontext.ProjectFileName)
+				} else {
+					if svc.RelativePath != "" {
+						svc.RelativePath = filepath.Join(svc.RelativePath, projectFile)
+					} else {
+						svc.Project.Path = filepath.Join(svc.Project.Path, projectFile)
+					}
+				}
+			}
+		}
 	}
```

Comment:
This doesn't sit right with me.  We shouldn't really have any language/service checks etc within the generic project configuration. 

I would prefer to see something split out where we call down into the underlying language service where a special case scenario can be evaluated to hydrate service configuration , etc.

Reply:
> I would prefer to see something split out where we call down into the underlying language service where a special case scenario can be evaluated to hydrate service configuration , etc.

Yeah, that makes sense.  I'll take a look and build something in that direction.

Reply:
Took a run at this with [e98cda2](https://github.com/Azure/azure-dev/pull/616/commits/e98cda2e5d038bc84347ca9288a023c58a2cf0ff), let me know if this is what you were thinking, @wbreza.

Reply:
Turns out this approach did not work (I tried to move the logic in to the NewDotNetService method and then have show call GetProject() and walk over that structure) because GetProject fails when infrastructure has not been provisioned (since it tries to find the target resource for each service, and of course they don't exist if you haven't run provision).

After talking with @wbreza about some possible fixes, we both decided the simplest solution for now is to just move this logic into `show` itself.  We both agree that it would be nice for GetProject to work even when infrastructure has not been provisioned (or some middle ground between ProjectConfig and Project that has this information but hasn't resolve resources yet), but that's a larger work item with deeper implications across the system.

For now, we'll just go with the logic living in `show`.